packages/frontend/src/components/wallet/TokenAmount.js

```diff
@@ -28,21 +31,37 @@ const showFullAmount = (amount, decimals, symbol) =>
         ? `${formatTokenAmount(amount, decimals, decimals)} ${symbol}`
         : '';
 
-const TokenAmount = ({ token: { balance, onChainFTMetadata }, withSymbol = false, className, showFiatAmount = true, "data-test-id": testId }) => (
-    <div className={className} title={showFullAmount(balance, onChainFTMetadata?.decimals, onChainFTMetadata?.symbol)} data-test-id={testId}>
-        <div>
-            {balance
-                ? formatToken(balance, onChainFTMetadata?.decimals)
-                : <span className='dots' />
+const TokenAmount = ({ 
+    token: { balance, onChainFTMetadata, priceInUSD, isWhiteListed = true }, 
+    withSymbol = false, 
+    className, 
+    showFiatAmount = true, 
+    "data-test-id": testId 
+}) => {
+    const tokenBalance = balance && formatToken(balance, onChainFTMetadata?.decimals);
+    const tokenPrice = (tokenBalance && priceInUSD) && big(+tokenBalance * +priceInUSD).toFixed(2);
```

Comment:
1. No need to use `big` here; these numbers aren't likely to exceed JS's native number type's boundaries.
2. Since we _always_ display these values to 2 decimal places, it would be nice to either store them formatted that way in our redux store, or format them for display when selecting the values from the `tokenFiatValues` reducer so that we don't have to sprinkle `toFixed(2)` throughout our components everywhere that we consume the values and if we later want to change the decimal accuracy we're displaying to, we don't need to find multiple places where we are formatting the value for display and update them all.

Reply:
Storing the formatted values in the store might do some issues in future calculations. If you want, I will also change this. But I prefer to work with original data and format it when we need this. All values in this case will be a little bit pure and close to the truth. Anyway, we have to decide how to modify `tokenFiatValues`.